### PR TITLE
NOISSUE - Oauth2.0 blog post

### PR DESCRIPTION
**Pull request by rodneyosodo**
### What does this do?
Submit blogpost for review

### Which issue(s) does this PR fix/relate to?
No issue

### List any changes that modify/break current functionality
Magistrala OAuth2.0 blogpost

### Have you included tests for your changes?
No

### Did you document any new/modified functionality?
Yes

### Notes
Not to be merged but to be reviewed for publishing to https://medium.com/abstract-machines-blog

### CODE REVIEW

File: docs/oauth.md [hunk 1]
```diff
@@ -0,0 +1,463 @@
+# Integrating OAuth2.0 with Magistrala
+
+Over the past months, we have been working on integrating OAuth2.0 with Magistrala. We are happy to announce that we have completed the integration and it is [now available](https://github.com/absmach/magistrala/pull/2103). We believe that this will open up a lot of possibilities for magistrala and we are very excited about the future of magistrala. We are planning to add more features to the OAuth2.0 integration in future releases. We are also planning to add support for more OAuth2.0 providers. This will enable users to use their preferred OAuth2.0 provider to authenticate with magistrala.
```
Comment (Musilah) [inline docs/oauth.md]: Capitalize the first letter of Magistrala

File: docs/oauth.md [hunk 2]
```diff
@@ -0,0 +1,463 @@
...
+6. The resource server validates the access token, and if valid, serves the request.
+7. The client receives the protected resource.
+
+This flow is demonstrated in the following diagram:
+
+![Generic OAuth2.0 flow](img/genericflow.png)
+
+Magistrala can now be the resource server and Google as one of the authorization servers. We have implemented the OAuth2.0 [authorization code flow](https://datatracker.ietf.org/doc/html/rfc6749#section-4.1). The authorization code flow is used to obtain an access and refresh token to authorize API requests. The UI client initiates the flow by redirecting the user to the authorization server(google). The user authenticates and authorizes the client(Magistrala). The authorization server(google) redirects the user back to the client(Magistrala users service) with an authorization code. The client(Magistrala users service) exchanges the authorization code for an access token and a refresh token. The access token is used to authenticate API requests(get user details). The refresh token is used to obtain a new access token when the current access token becomes invalid or expires. This flow is demonstrated in the following diagram:
```
Comment (Musilah) [inline docs/oauth.md]: Use Google instead of google.

File: docs/oauth.md [hunk 3]
```diff
@@ -0,0 +1,463 @@
...
+
+![authorization code flow](img/codeflow.png)
+_The authorization code flow(from https://medium.com/javarevisited/oauth-2-0-authorization-code-flow-in-spring-boot-d8ff393f316d)_
+
+There are different grant types in OAuth2.0. The grant type is a string representing the authorization grant type that the client is using to request the access token. The grant type is included in the request to the token endpoint. The grant type is used to specify the method of obtaining the access token. The grant type is a required parameter in the request to the token endpoint. The grant types are:
```
Comment (Musilah) [inline docs/oauth.md]: The grant type has been repeated at the beginning of four sentences concurrently.

File: docs/oauth.md [hunk 4]
```diff
@@ -0,0 +1,463 @@
...
+   Internally we prefix the state with `signin` or `signup` to differentiate between the sign-in and sign-up flows. This is to identify the flow that the user is in. This is important because the sign-in and sign-up flows are different and we need to handle them differently.
+
+4. You can customize the consent screen by providing the application name and the authorized domains. You can also provide the application logo and the application homepage. You can find the consent screen in the [OAuth consent screen section](https://console.cloud.google.com/apis/credentials/consent) of the project.
+
+More information about the OAuth2.0 authorization code flow can be found [here](https://developers.google.com/identity/openid-connect/openid-connect).
+
+Here is an example of the OAuth2.0 authorization code flow with Google as the authorization server:
+
+<iframe width="560" height="315" src="https://www.youtube.com/embed/DxhqnHkHolU?si=BS5PJn8Qevj8Dp1j" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe>
```
Comment (Musilah) [inline docs/oauth.md]: This image does not show up locally

Comment (rodneyosodo) [reply docs/oauth.md]: Are you able to see this https://youtu.be/DxhqnHkHolU?si=72cOw9wd1pWNcDQ6?

**Comment by dborovcanin** [comment]
@rodneyosodo Can you create a separate `blogs` directory so we have them nicely organized?

**Review by dborovcanin — APPROVED** [review]

**Pull request merged**